hardware_interface/test/test_resource_manager.cpp

```diff
@@ -1329,3 +1329,238 @@ TEST_F(TestResourceManager, managing_controllers_reference_interfaces)
   EXPECT_THROW(
     rm.make_controller_reference_interfaces_unavailable("unknown_controller"), std::out_of_range);
 }
+
+TEST_F(TestResourceManager, handle_error_on_hardware_read_and_write)
+{
+  // values to set to hardware to simulate failure on read and write
+  static constexpr double READ_FAIL_VALUE = 28282828.0;
+  static constexpr double WRITE_FAIL_VALUE = 23232323.0;
+
+  rclcpp_lifecycle::State state_active(
+    lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE,
+    hardware_interface::lifecycle_state_names::ACTIVE);
+
+  hardware_interface::ResourceManager rm(ros2_control_test_assets::minimal_robot_urdf, false);
+  activate_components(rm);
+
+  auto status_map = rm.get_components_status();
+  EXPECT_EQ(
+    status_map[TEST_ACTUATOR_HARDWARE_NAME].state.id(),
+    lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE);
+  EXPECT_EQ(
+    status_map[TEST_SYSTEM_HARDWARE_NAME].state.id(),
+    lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE);
+  EXPECT_EQ(
+    status_map[TEST_SENSOR_HARDWARE_NAME].state.id(),
+    lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE);
+
+  auto claimed_itf_actuator =
+    rm.claim_command_interface(TEST_ACTUATOR_HARDWARE_COMMAND_INTERFACES[0]);
+  auto claimed_itf_system = rm.claim_command_interface(TEST_SYSTEM_HARDWARE_COMMAND_INTERFACES[0]);
+
+  const auto time = rclcpp::Time(0);
+  const auto duration = rclcpp::Duration::from_seconds(0.01);
+
+  // check if all interfaces are available
+  auto interfaces_available = [&](const auto actuator_interfaces, const auto system_interfaces)
```

Comment:
```suggestion
  auto interfaces_available = [&](const bool actuator_interfaces, const bool system_interfaces)
```
it's useful to remind ppl that these are boolean flags

Reply:
Done